esx_service/vmdk_ops.py

```diff
@@ -668,20 +670,27 @@ def vm_uuid2name(vm_uuid):
         return None
     return vm.config.name
 
-# Return error, or None for OK.
-def attachVMDK(vmdk_path, vm_uuid):
-    vm = findVmByUuid(vm_uuid)
-    logging.info("*** attachVMDK: %s to %s VM uuid = %s",
-                 vmdk_path, vm.config.name, vm_uuid)
-    return disk_attach(vmdk_path, vm)
 
+def attach_or_detach_VMDK(cmd, vmdk_path, vm_name, bios_uuid, vc_uuid):
```

Comment:
This is better. Here's what I have in my mind:

`
# Return error, or None for OK.
def attachVMDK(vmdk_path, vm_name, bios_uuid, vc_uuid):
    logging.info("*** attachVMDK: %s to %s VM uuid = %s",
                vmdk_path, vm.config.name, vm_uuid)
    vm = findVm(vmdk_path, vm_name, bios_uuid, vc_uuid)
    return disk_attach(vmdk_path, vm)

# Return error, or None for OK.
def detachVMDK(vmdk_path, vm_name, bios_uuid, vc_uuid):
    logging.info("*** detachVMDK: %s from %s VM uuid = %s",
                vmdk_path, vm.config.name, vm_uuid)
    vm = findVm(vmdk_path, vm_name, bios_uuid, vc_uuid)
    return disk_detach(vmdk_path, vm)

def findVm(vmdk_path, vm_name, bios_uuid, vc_uuid):
    logging.info("*** findVm: vmdk_path=%w, vm_name=%s, bios_uuid=%s, vc_uuid=%s)",
                 vmdk_path, vm_name, bios_uuid, vc_uuid)
    vm = None
    if vc_uuid:
        vm = findVmByUuid(vc_uuid)
    if not vm: # either vc_uuid is not even passed, or we failed to find the VM by VC uuid:
        logging.warning("Failed to find VM by VC UUID %s, trying BIOS UUID %s", vc_uuid, bios_uuid)
        vm = findVmByUuid(bios_uuid)
    if not vm: # can find VM by VC or BIOS uuid
        msg = "Failed to find VM object for %s (bios %s vc %s)" % (vm_name, bios_uuid, vc_uuid)
        logging.error(msg)
        return err(msg)

    if vm.config.name != vm_name:
        logging.warning("vm_name from vSocket '%s' does not match VM object '%s' ", vm_name, vm.config.name)

    return vm
`

Not really a big concern. So you can take the call. Thanks for addressing this!